plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -929,26 +930,32 @@ func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods [
 	if len(term.TopologyKey) == 0 {
 		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
 	}
-	matchingPodExists := false
+
 	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
 		return false, false, err
 	}
+
+	podNum := int32(1)
 	for _, existingPod := range allPods {
 		match := priorityutil.PodMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector)
 		if match {
-			matchingPodExists = true
```

Comment:
@k82cn Please look at this function as an independent module, meaning that it should work  as its name and comment imply. The function is called `anyPodMatchesPodAffinityTerm` and it is in fact used for checking both affinity and anti-affinity. Put yourself in the shoes of someone who will read this code in the future. It will be hard for someone to understand the logic.
I think you should write the logic for checking affinity, and use logical negation of the return value for anti-affinity. This is how the current code is written.